api/project/docker-compose.yml

```diff
@@ -0,0 +1,28 @@
+version: '3.1'
+
+services:
+  backend:
+    platform: linux/amd64
+    build:
+      context: .
+      dockerfile: ./Dockerfile
+    ports:
+      - 3000:3000
+    env_file:
+      - ./.env
+    links:
+      - db
+    depends_on:
+      - db
+    container_name: backend
+
+  db:
+    platform: linux/amd64
+    image: mysql:8
+    ports:
+      - 3306:3306
+    env_file:
+      - ./.env
+    volumes:
+      - ./init.sql:/docker-entrypoint-initdb.d/init.sql
```

Comment:
https://orkhan.gitbook.io/typeorm/docs/connection-options#connection-options-example
> `synchronize` - Indicates if database schema should be auto created on every application launch. Be careful with this option and don't use this in production - otherwise you can lose production data. This option is useful during debug and development. As an alternative to it, you can use CLI and run schema:sync command.

typeorm에는 syncronize라는 옵션이 있습니다. 이거 이용하시면 이렇게 직접 마이그레이션 스크립트 작성하지 않으셔도 됩니다. 물론 의도치 않은 마이그레이션에 의해 데이터가 손실될 수 있어서 프로덕션에서는 안 쓰는게 좋구요, 개발 환경에서는 진짜 훨~~~~씬 편하니까 쓰시는 걸 추천드릴게요. .env에 포함시키셔서 prod, dev 환경에 syncronize옵션이 다르게 들어갈 수 있도록 하시면 될 것 같아요.

장고 써보셔서 아시겠지만 orm 쓸 때 마이그레이션 관리하는게 여간 쉬운 일이 아니죠 ㅎㅎ 그래서 인지 이런 기능이 최근에 추가되었더라구요... nestjs, typeorm 쓴 레포들에서 마이그레이션 스크립트가 없는 이유이기도 합니다.

Reply:
오홍 감사합니다 그럼 production 환경에서는 스크립트가 필요하긴 한 건가요? 아니면 또 다른 방법을 쓰시나요?